File: rb/features.rb
# FantasyRolePlaying SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module FantasyRolePlayingFeatures
  def self.make_feature(name)
    case name
    when "base"
      FantasyRolePlayingBaseFeature.new
    when "ratelimit"
      FantasyRolePlayingRatelimitFeature.new
    when "retry"
      FantasyRolePlayingRetryFeature.new
    when "test"
      FantasyRolePlayingTestFeature.new
    when "timeout"
      FantasyRolePlayingTimeoutFeature.new
    else
      FantasyRolePlayingBaseFeature.new
    end
  end
end

File: py/features.py
# FantasyRolePlaying SDK feature factory

from feature.base_feature import FantasyRolePlayingBaseFeature
from feature.test_feature import FantasyRolePlayingTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FantasyRolePlayingBaseFeature(),
        "test": lambda: FantasyRolePlayingTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
